xclim/indices/_agro.py

```diff
@@ -507,6 +508,133 @@ def cool_night_index(
     return cni
 
 
+@declare_units(pr="[precipitation]", evspsblpot="[precipitation]")
+def dryness_index(
+    pr: xarray.DataArray,
+    evspsblpot: xarray.DataArray,
+    lat: xarray.DataArray | str | None = None,
+    wo: str = "200 mm",
+    freq: str = "YS",
+) -> xarray.DataArray:
+    """Dryness Index.
+
+    Mean minimum temperature for September (northern hemisphere) or March (Southern hemisphere).
+    Used in calculating the Géoviticulture Multicriteria Classification System.
+
+    Warnings
+    --------
+    Dryness Index expects CF-Convention conformant potential evapotranspiration (positive up).
+
+    Parameters
+    ----------
+    pr : xarray.DataArray
+        Precipitation.
+    evspsblpot : xarray.DataArray
+        Potential evapotranspiration.
+    lat : xarray.DataArray or {"north", "south"}, optional
+        Latitude coordinate as an array, float or string.
+        If None, a CF-conformant "latitude" field must be available within the passed DataArray.
+    wo : str
+        The initial soil water reserve accessible to root systems.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray, [mm]
+        Absolute Dryness Index values.
+
+    Notes
+    -----
+    TODO: Describe this indice.
+
+    References
+    ----------
+    :cite:cts:`tonietto_multicriteria_2004,riou_determinisme_1994`
+
+    """
+    if parse_offset(freq) != (1, "A", True, "JAN"):
+        raise ValueError(f"Freq not allowed: {freq}. Must be `YS` or `AS-JAN`")
+
+    # Resample all variables to monthly totals in mm units.
+    evspsblpot = rate2amount(evspsblpot, out_units="mm").resample(time="MS").sum()
+    pr = rate2amount(pr, out_units="mm").resample(time="MS").sum()
+    wo = convert_units_to(wo, "mm")
+
+    # Different potential evapotranspiration rates for northern hemisphere and southern hemisphere.
+    # wo_adjustment is the initial soil moisture rate at beginning of season.
+    adjustment_array_north = xarray.DataArray(
+        [0, 0, 0, 0.1, 0.3, 0.5, 0.5, 0.5, 0.5, 0, 0, 0],
+        dims="month",
+        coords=dict(month=np.arange(1, 13)),
+    )
+    adjustment_array_south = xarray.DataArray(
+        [0.5, 0.5, 0.5, 0.5, 0, 0, 0, 0, 0, 0, 0.1, 0.3],
+        dims="month",
+        coords=dict(month=np.arange(1, 13)),
+    )
```

Comment:
The 2004 paper says: 
```
In the Southern Hemisphere, k = 0.1 for October,
0.3 for November and 0.5 for the months from De-
cember to March.
```
whereas your array implies this starts in November. Is this just a typo, or maybe I don't understand correctly what `di_south` does below?

Reply:
It's a typo, for sure. I likely read the paper wrong. Will revise.